src/index.js

```diff
@@ -167,7 +172,9 @@ module.exports = function (babel) {
               let binding = path.scope.getBinding(local.name);
 
               binding.referencePaths.forEach((referencePath) => {
-                referencePath.replaceWith(getMemberExpressionFor(global));
+                if (!isTSQualifiedName(referencePath.parentPath)) {
```

Comment:
This is definitely the right direction, but I do not believe this is the full extent of the fix. While this fixes the error in ember-data, I’m seeing errors in other projects that use TypeScript for types **other than** `TSQualifiedName`.

I think we need to detect _any_ type specific node and skip, not just TSQualifiedName.

Reply:
Another common one I’ve seen is `TSDeclareFunction`. 


https://github.com/emberjs/ember.js/issues/18995

Reply:
I had wondered about this as well! I updated the PR to check if the `type` starts with `TS` since it looks like all typescript types are prefixed with `TS`. This doesn't feel great but I don't see a way to get a list of all the typescript types dynamically, and doing something like the `starts with TS` check should be a guard against any AST types that babel/typescript adds in the future. What do you think about that approach?

Reply:
Ya, seems good to me, though I agree that it does seem a bit unfortunate that there isn't something better.